docs/modules/SDDs/pages/0024-naming-scheme-v2.adoc

```diff
@@ -0,0 +1,76 @@
+= SDD 0024 - Project Syn Tools Naming Scheme v2
+
+:sdd_author:    Tobias Brunner
+:sdd_owner:
+:sdd_reviewers:
+:sdd_date:      2020-10-23
+:sdd_status:    draft
+include::partial$meta-info-table.adoc[]
+
+[NOTE]
+.Summary
+====
+Naming is hard, this page helps to make it easy. It describes the rules to name Project Syn tools.
+====
+
+== Motivation
+
+In xref:0002-naming-scheme.adoc[SDD 0002 - Naming Scheme] a naming scheme was defined for Project Syn tools which turned out to be sub-optimal.
+Having tools with non-descriptive names increases the (already relatively high) barrier of entry into the world of Project Syn.
+This SDD describes a more descriptive naming scheme which solves this issue.
+
+=== Goals
+
+* Naming scheme which is descriptive
+* Applicable to all new tools from now on
+
+=== Non-Goals
+
+* Renaming of existing tools (like Lieutenant, Steward and Commodore) as they will change over time anyways, ultimately being replaced by newer tools.
+
+== Design Proposal
+
+Tools must have descriptive display names of the form `Syn ${type}[ ${purpose}]`.
+By default the binary name for a tool would be `syn-$type[-$purpose]`, but deviations are allowed, see below.
+
+`$type`:: Describes the kind of the tool.
++
+Examples for types:
+
+* Operator
+* Agent
+* CLI
+
+`$purpose`:: Optional to describe purpose of the tool
+
+Examples:
+
+* syn-cli-compile
+* syn-operator-tenant
+* syn-agent
+
+=== Allowed deviations between tool and binary name
+
+A tool's binary is allowed to deviate as long as it's discoverable with `syn<Tab><Tab>`, and the suffix is well known as a particular type of tool.
+
+Examples where deviation is OK:
+
+* The Syn Catalog Compiler, could have the binary name `syncc`, because nobody wants to type `syn-catalog-compiler`, and `cc` is well known as indicating a compiler (technically `cc` is the suffix indicating a C compiler, but luckily catalog also starts with a c).
```

Comment:
```suggestion
* The Syn Catalog Compiler could have the binary name `syncc`, because nobody wants to type `syn-catalog-compiler`, and `cc` is well known as indicating a compiler (technically `cc` is the suffix indicating a C compiler, but luckily catalog also starts with a c).
```